pkg/podmounter/mountoptions/mount_options.go

```diff
@@ -0,0 +1,130 @@
+func Send(ctx context.Context, sockPath string, options Options) error {
+	message, err := json.Marshal(&options)
+	if err != nil {
+		return fmt.Errorf("failed to marshal message to send %s: %w", sockPath, err)
+	}
+
+	conn, err := net.Dial("unix", sockPath)
+	if err != nil {
+		return fmt.Errorf("failed to dial to unix socket %s: %w", sockPath, err)
+	}
+	defer conn.Close()
+
+	unixConn := conn.(*net.UnixConn)
+
+	unixRights := syscall.UnixRights(options.Fd)
+	messageN, unixRightsN, err := unixConn.WriteMsgUnix(message, unixRights, nil)
+	if err != nil {
+		return fmt.Errorf("failed to write to unix socket %s: %w", sockPath, err)
+	}
+	if len(message) != messageN || len(unixRights) != unixRightsN {
+		return fmt.Errorf("partial write to unix socket %s: message: size %d - written %d, unix rights: size %d - written %d",
+			sockPath, len(message), messageN, len(unixRights), unixRightsN)
+	}
+
+	return nil
+}
+
+var (
+	messageRecvSize = 1024
+	// We only pass one file descriptor and its 32 bits
+	unixRightsRecvSize = syscall.CmsgSpace(4)
+)
+
+// Recv receives passed mount options via `Send` function through given `sockPath`.
+func Recv(ctx context.Context, sockPath string) (Options, error) {
+	l, err := net.Listen("unix", sockPath)
+	if err != nil {
+		return Options{}, fmt.Errorf("failed to listen unix socket %s: %w", sockPath, err)
+	}
+	defer l.Close()
+
+	conn, err := l.Accept()
+	if err != nil {
+		return Options{}, fmt.Errorf("failed to accept connection from unix socket %s: %w", sockPath, err)
+	}
+
+	unixConn := conn.(*net.UnixConn)
+
+	messageBuf := make([]byte, 0)
+	unixRightsBuf := make([]byte, 0)
+
+	// Read in a loop to consume the whole message
+	for {
+		message := make([]byte, messageRecvSize)
+		unixRights := make([]byte, unixRightsRecvSize)
+
+		messageN, unixRightsN, _, _, err := unixConn.ReadMsgUnix(message, unixRights)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+
+			return Options{}, fmt.Errorf("failed to read message from unix socket %s: %w", sockPath, err)
+		}
+
+		messageBuf = append(messageBuf, message[:messageN]...)
+		unixRightsBuf = append(unixRightsBuf, unixRights[:unixRightsN]...)
+	}
+
+	var options Options
+	err = json.Unmarshal(messageBuf, &options)
+	if err != nil {
+		return Options{}, fmt.Errorf("failed to decode mount options from unix socket %s: %w", sockPath, err)
+	}
+
+	fds, err := parseUnixRights(unixRightsBuf)
+	if err != nil {
+		return Options{}, fmt.Errorf("failed to decode unix rights from unix socket %s: %w", sockPath, err)
+	}
+
+	if len(fds) != 1 {
+		return Options{}, fmt.Errorf("expected to got one file descriptor from unix socket %s, but got %d", sockPath, len(fds))
+	}
+
+	options.Fd = fds[0]
+	return options, nil
+}
+
+// parseUnixRights parses given socket control message to extract passed file descriptors.
+func parseUnixRights(buf []byte) ([]int, error) {
+	socketControlMessages, err := syscall.ParseSocketControlMessage(buf)
```

Comment:
I'm not sure what this is meant to do

Reply:
as the comment suggests "... parses given socket control message to extract passed file descriptors.". Maybe I can add references to "Ancillary messages" and "SCM_RIGHTS" from https://man7.org/linux/man-pages/man7/unix.7.html.